tinkerpop3/src/main/java/io/shiftleft/overflowdb/structure/NodeRef.java

```diff
@@ -5,28 +5,112 @@
 import org.apache.tinkerpop.gremlin.structure.Graph;
 import org.apache.tinkerpop.gremlin.structure.Vertex;
 import org.apache.tinkerpop.gremlin.structure.VertexProperty;
-import org.apache.tinkerpop.gremlin.structure.util.StringFactory;
 
 import java.io.IOException;
 import java.util.Iterator;
 
-public abstract class NodeRef<V extends Vertex> extends ElementRef<V> implements Vertex {
+/**
+ * Wrapper for a node, which may be set to `null` by @ReferenceManager to avoid OutOfMemory errors.
+ * When it's cleared, it will be persisted to an on-disk storage.
+ */
+public abstract class NodeRef<N extends OverflowDbNode> implements Vertex {
+  public final long id;
 
-  public NodeRef(final Object vertexId, final Graph graph, V vertex) {
-    super(vertexId, graph, vertex);
+  protected final OverflowDbGraph graph;
+  protected N node;
+  private boolean removed = false;
```

Comment:
Do we still need the removed field? I think that is a left over from tinker world right?

Reply:
true, good point, done